include/materials/material_utility.h

```diff
@@ -0,0 +1,71 @@
+#ifndef MPM_MATERIAL_UTILITY_H_
+#define MPM_MATERIAL_UTILITY_H_
+
+#include <cmath>
+
+#include "data_types.h"
+
+namespace mpm {
+namespace materials {
+
+//! Compute mean stress p (tension positive)
+//! \param[in] stress Stress in Voigt notation where positive is tension
+inline const double p(const Eigen::Matrix<double, 6, 1>& stress);
```

Comment:
Please avoid returning by const value for built-in data types. 
```you are promising that the original variable (inside the function frame) will not be modified. And again, because you’re returning it by value, it’s copied so the original value could never be modified via the return value. For built-in types, it doesn’t matter whether you return by value as a const, so you should avoid confusing the client programmer and leave off the const when returning a built-in type by value.```

For more info: https://www.linuxtopia.org/online_books/programming_books/thinking_in_c++/Chapter08_014.html

Reply:
Keep const for `Eigen` data types

Reply:
I see. Understood.